backend/src/api/mutation.rs

```diff
@@ -242,37 +237,31 @@ impl Mutation {
         realms: Vec<RealmLineageComponent>,
         context: &Context,
     ) -> ApiResult<CreateRealmLineageOutcome> {
-        if context.auth != AuthContext::TrustedExternal {
-            return Err(not_authorized!("only trusted external applications can use this mutation"));
-        }
-
-        if realms.len() == 0 {
-            return Ok(CreateRealmLineageOutcome { num_created: 0 });
-        }
-
-        if context.config.general.reserved_paths().any(|r| realms[0].path_segment == r) {
-            return Err(invalid_input!(key = "realm.path-is-reserved", "path is reserved and cannot be used"));
-        }
+        Realm::create_lineage(realms, context).await
+    }
 
-        let mut parent_path = String::new();
-        let mut num_created = 0;
-        for realm in realms {
-            let sql = "\
-                insert into realms (parent, name, path_segment) \
-                values ((select id from realms where full_path = $1), $2, $3) \
-                on conflict do nothing";
-            let res = context.db.execute(sql, &[&parent_path, &realm.name, &realm.path_segment])
-                .await;
-            let affected = map_db_err!(res, {
-                if constraint == "valid_path" => invalid_input!("path invalid"),
-            })?;
-            num_created += affected as i32;
+    /// Stores series information in Tobira's DB, so it can be mounted without having to be harvested first.
+    async fn announce_series(series: NewSeries, context: &Context) -> ApiResult<Series> {
+        Series::announce(series, context).await
+    }
 
-            parent_path.push('/');
-            parent_path.push_str(&realm.path_segment);
-        }
+    /// Adds a series block to an empty realm, if the series is known to Tobira.
```

Comment:
```suggestion
    /// Adds a series block to an empty realm and makes that realm derive its name from said series.
```

The "if the series is known" is not that important IMO, thats kind of implied? The same is true for the realm after all. And I added information about name derive stuff.